src/api/IntentResolution.ts

```diff
@@ -3,23 +3,44 @@
  * Copyright 2019 FINOS FDC3 contributors - see NOTICE file
  */
 
+import { Context } from '../context/ContextTypes';
 import { TargetApp } from './Types';
 
 /**
  * IntentResolution provides a standard format for data returned upon resolving an intent.
  * ```javascript
  * //resolve a "Chain" type intent
- * var intentR = await agent.raiseIntent("intentName", context);
- * //resolve a "Client-Service" type intent with data response
- * var intentR = await agent.raiseIntent("intentName", context);
- * var dataR = intentR.data;
+ * let resolution = await agent.raiseIntent("intentName", context);
+ *
+ * //resolve a "Client-Service" type intent with a data response
+ * let resolution = await agent.raiseIntent("intentName", context);
+ * try {
+ * 	   const result = await resolution.getData();
+ *     if (result) {
+ *         console.log(`${resolution.source} returned ${JSON.stringify(result)}`);
+ *     } else {
+ *         console.error(`${resolution.source} didn't return data`
+ *     }
+ * } catch(error) {
+ *     console.error(`${resolution.source} returned an error: ${error}`);
+ * }
  * ```
  */
 export interface IntentResolution {
+  /**
+   * The application that resolved the intent.
+   */
   readonly source: TargetApp;
   /**
-   * @deprecated not assignable from intent listeners
+   * The version number of the Intents schema being used.
+   */
+  readonly version?: string;
+  /**
+   * Retrieves a promise that will resolve to data returned by the
+   * application that resolves the raised intent. The promise MUST
+   * reject with a string from the `DataError` enumeration if an error
+   * is thrown by the intent handler, it rejects the returned promise,
+   * or it does not return a promise.
    */
-  readonly data?: object;
-  readonly version: string;
+  getData(): Promise<Context>;
```

Comment:
As discussed, couldn't we rename this field ```getResult()``` as it would allow to handle potential future use cases where the result is not a data structure (cf. discussions on notifications)

That said, I don't think we need to have a method here (i.e. a Promise would be enough as you suggested). I guess most implementation will keep the promise as a private hidden field that would be returned by getData/getResult method. Indeed we don't want to keep the data in the Agent until the method is called - so the agent is likely to send the result once received and store it in a private field in the client structure to make it available through getData... (so there is no point in this extra complexity..)

Reply:
My only concern on using `getResult()` is that it will require a union return type (`Promise<Context | Channel>`), typescript is good at those, but other languages are not and although we use Typescript to define the API it should ideally be implementable in other languages. 

I would tend to agree on a promise vs. a function returning a promise. Using a function does mean either retaining the data in the desktop agent service or a hidden field, as opposed to it living in the promise itself. However, I have had to demonstrate to a couple of people recently that you can `await` promises, not just functions returning promises, so I was willing to accept that developers might be more familiar with calling a function.

Reply:
Hmm, we also need that union type as the return type for the IntentHandler:
```typescript
export type IntentHandler = (context: Context) => Promise<Context | PrivateChannel> | void;
```

Perhaps we should introduce an explicate union type (as we do for [`TargetApp`](https://fdc3.finos.org/docs/api/ref/Types#targetapp)). While that doesn't change anything in Typescript, it could be a class in other languages that can hold either response type?

Reply:
Updated to getResult() (which will return a union type in a later PR when we add support for returning a channel)